src/main/scala/chisel3/simulator/PeekPokeAPI.scala

```diff
@@ -126,13 +133,25 @@ trait PeekPokeAPI {
       expected:     T,
       encode:       (Simulation.Value) => T,
       buildMessage: (T, T) => String
+    )(
+      implicit sourceInfo: SourceInfo
```

Comment:
```suggestion
     sourceInfo: SourceInfo
```
> Minor API design question: do we ever expect testableData.expect (with expected, encode and buildMessage arguments) to be called by a test user directly? If not, I have a slight preference for making the sourceInfo argument non-implicit there, and only having it implicit in the user-facing calls on SimulationData.

I agree with your assessment, this isn't really a user API so it should be explicit. However, it is a public API that power users could be using to implement their own peek/poke/expect API, so on backport, we can do a deprecation of the old API.